### PR TITLE
Core: Workaround for importlib incompatibility with python 2.6; Closes #875

### LINKED ISSUE #875: importlib not python 2.6 compatible
Motivation
----------
For the dynamic lfn2pfn function the module `importlib` is used. However, importlib is not part of python 2.6, thus it breaks the clients used on the ATLAS local root base. One option would be to use the backport of importlib for python 2.6 (on pypi) - needs to be investigated.

https://pypi.python.org/pypi/importlib

* Adding importlib to the dependencies works for python 2.6 and 2.7, but a install on python 3.x fails like this. 

For now we move the import to a conditional part of the code, as no module gets dynamically loaded at the moment anyway; This needs to be addressed later on with the dynamic loading of permissions and policies, as this should be handled in a similar way.

### PR DESCRIPTION
**Pull request by bari12**
Core: Workaround for importlib incompatibility with python 2.6; Closes #875

### CODE REVIEW

File: lib/rucio/rse/protocols/protocol.py [hunk 1]
```diff
@@ -143,7 +143,13 @@ def _module_init_(cls):
         except (NoOptionError, NoSectionError):
             pass
         if policy_module:
-            importlib.import_module(policy_module)
+            # TODO: The import of importlib is done like this due to a depdency issue with python 2.6 and incompatibility of the module with py3.x
```
Comment (vingar) [inline lib/rucio/rse/protocols/protocol.py]: depdency -> dependency

File: lib/rucio/rse/protocols/protocol.py [hunk 2]
```diff
@@ -143,7 +143,13 @@ def _module_init_(cls):
...
+            # More information https://github.com/rucio/rucio/issues/875
+            try:
+                import importlib
+                importlib.import_module(policy_module)
+            except:
+                pass
```
Comment (vingar) [inline lib/rucio/rse/protocols/protocol.py]: should it fail silently or with:
`raise exception.MissingDependency('Missing dependency : importlib')` ?

File: lib/rucio/rse/protocols/protocol.py [hunk 3]
```diff
@@ -143,7 +143,13 @@ def _module_init_(cls):
         except (NoOptionError, NoSectionError):
             pass
         if policy_module:
-            importlib.import_module(policy_module)
+            # TODO: The import of importlib is done like this due to a depdency issue with python 2.6 and incompatibility of the module with py3.x
+            # More information https://github.com/rucio/rucio/issues/875
+            try:
+                import importlib
+                importlib.import_module(policy_module)
+            except:
```
Comment (vingar) [inline lib/rucio/rse/protocols/protocol.py]: except ImportError:

File: lib/rucio/rse/protocols/protocol.py (hunk 2 above)
Comment (bari12) [reply lib/rucio/rse/protocols/protocol.py]: Right, it should just raise it through...

**Review by vingar — APPROVED** [review]

**Pull request merged**